python/modules/EventInfo.py

```diff
@@ -13,35 +13,56 @@
 class EventInfo(Module):
     def __init__(
         self,
-        storeVariables = []
+        storeVariables = [],
+        useExistingSum = False
     ):
         self.storeVariables = storeVariables
+        self.useExistingSum = useExistingSum
 
     def beginJob(self):
         pass
     def endJob(self):
         pass
     def beginFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
         self.nGenWeights = 0
+        self.nGenEventCount = 0
+        self.genEventSumw2 = 0
+        self.LHEScaleSumw = [0, 0, 0, 0, 0, 0, 0, 0, 0]
+        self.existingSumSaved = False
         self.out = wrappedOutputTree
         for variable in self.storeVariables:
             variable[0](self.out)
         
     def endFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
     	if not Module.globalOptions["isData"]:
+            nGenEventCount_parameter = ROOT.TParameter(float)("genEventCount", self.nGenWeights)
             nGenWeight_parameter = ROOT.TParameter(float)("sumGenWeights", self.nGenWeights)
+            genEventSumw2_parameter = ROOT.TParameter(float)("sumGenWeights2", self.genEventSumw2)
```

Comment:
why do you only take the sumw2 and not also the sumw?

Reply:
`self.nGenWeights` is already sumw, see line 56.